pages/index.js

```diff
@@ -23,6 +25,26 @@ class Index extends Component {
   componentDidMount() {
     connectSocket()
     this.props.dispatch(projectsListener())
+    this.updateStateFromQueries()
+  }
+
+  componentDidUpdate(previousProps) {
+    const showFailed = this.props.filters.showFailed
+    const projectQuery = this.props.filters.projectQuery
+    const shouldSetQueryParameters = previousProps.filters.showFailed !== showFailed || previousProps.filters.projectQuery !== projectQuery
```

Comment:
I think this 'has anything changed' logic can be done more neatly (compare the whole `query` object you're going to push to? inside `setQueryParameters` then this code becomes really simple.